### PR TITLE
libzfs warning fixes

### PR DESCRIPTION
**Pull request by kev009**
Credit for this works belongs to Aaron Prieger, my summer FreeBSD kernel intern

I simply rebased on master and squashed as requested in https://github.com/openzfs/openzfs/pull/395.  If any other changes are needed I will make them.

### CODE REVIEW

**Comment by prakashsurya** [comment]
Thank you! I can look this over this coming week.

File: usr/src/cmd/zfs/zfs_main.c [hunk 1]
```diff
@@ -1322,7 +1322,7 @@ zfs_do_destroy(int argc, char **argv)
 			char buf[16];
 			zfs_nicenum(cb.cb_snapused, buf, sizeof (buf));
 			if (cb.cb_parsable) {
-				(void) printf("reclaim\t%llu\n",
+				(void) printf("reclaim\t%"PRIu64"\n",
 				    cb.cb_snapused);
```
Comment (ahrens) [inline usr/src/cmd/zfs/zfs_main.c]: The typical way of addressing this in ZFS has been to cast the argument to `long long`.  Personally, I find that more readable than `"PRIu64"`.

Comment (kev009) [reply usr/src/cmd/zfs/zfs_main.c]: It looks like a Hungarian notation. Both are specified in C99, do you want PRIu64 changed to long long casts?

Comment (ahrens) [reply usr/src/cmd/zfs/zfs_main.c]: Yes, thank you.

**Comment by kev009** [comment]
Going to just close this and wait for something to percolate from ZOL since Illumos' days seem pretty numbered with Delphix announcement.

**Comment by yuripv** [comment]
What is this about?

**Comment by ahrens** [comment]
FYI - I don't know anything about "illumos' days being numbered".  We (Delphix) are shipping products based on illumos and are working to ensure its continued long life.

**Comment by kev009** [comment]
Based on https://www.delphix.com/blog/kickoff-future-eko-2018

I'll wait a while to see how the community handles that transition

**Comment by pbhenson** [comment]
That does seem to imply that Delphix is transitioning from illumos to OpenZFS under Linux?

**Pull request closed without merging**